hlapipeline/alignimages.py

```diff
@@ -552,27 +599,63 @@ def update_image_wcs_info(tweakwcs_output,imagelist):
             image_name = imagelist[imgctr]
             if imgctr > 0: #close previously opened image
-                print("CLOSE {}".format(hdulist[0].header['FILENAME'])) #TODO: Remove before deployment
+                imageName = hdulist[0].header['FILENAME']
+                print("CLOSE {}\n".format(imageName)) #TODO: Remove before deployment
                 hdulist.flush()
                 hdulist.close()
+                out_image_list.append(imageName)
+                # Create headerlet
+                out_headerlet = headerlet.create_headerlet(imageName, hdrname=wcsName, wcsname=wcsName)
+
+                # TODO: First of two calls to Michele's update headerlet subroutine goes here!
+
+                if imageName.endswith("flc.fits"):
+                    headerlet_filename = imageName.replace("flc", "flt_hlet")
+                if imageName.endswith("flt.fits"):
+                    headerlet_filename = imageName.replace("flt", "flt_hlet")
+                out_headerlet.writeto(headerlet_filename,clobber=True)
+                print("Wrote headerlet file {}.\n\n".format(headerlet_filename))
+                out_headerlet_list.append(headerlet_filename)
             hdulist = fits.open(image_name, mode='update')
+            #generate wcs name for updated image header, headerlet
+            if not hdulist['SCI',1].header['WCSNAME'] or hdulist['SCI',1].header['WCSNAME'] =="": #Just in case header value 'wcsname' is empty.
+                wcsName = "FIT_{}".format(item.meta['catalog_name'])
+            else:
+                wcsName = "{}-FIT_{}".format(hdulist['SCI',1].header['WCSNAME'],item.meta['tweakwcs_info']['catalog'])
+
             sciExtDict = {}
             for sciExtCtr in range(1, amutils.countExtn(hdulist) + 1): #establish correct mapping to the science extensions
                 sciExtDict["{}".format(sciExtCtr)] = fileutil.findExtname(hdulist,'sci',extver=sciExtCtr)
             imgctr += 1
-        updatehdr.update_wcs(hdulist, sciExtDict["{}".format(item.meta['chip'])], item.wcs, wcsname='TWEAKDEV', reusename=True, verbose=True) #TODO: May want to settle on a better name for 'wcsname'
+        updatehdr.update_wcs(hdulist, sciExtDict["{}".format(item.meta['chip'])], item.wcs, wcsname=wcsName, reusename=True, verbose=True)
         print()
-    print("CLOSE {}".format(hdulist[0].header['FILENAME'])) #TODO: Remove before deployment
+    imageName = hdulist[0].header['FILENAME']
+    print("CLOSE {}\n".format(imageName)) #TODO: Remove before deployment
     hdulist.flush() #close last image
     hdulist.close()
+    out_image_list.append(imageName)
+    out_headerlet = headerlet.create_headerlet(imageName, hdrname=wcsName, wcsname=wcsName)
+
+    # TODO: Second of two calls to Michele's update headerlet subroutine goes here!
 
+    if imageName.endswith("flc.fits"):
+        headerlet_filename = imageName.replace("flc", "hlet")
```

Comment:
I would think this needs to be consistent with line 623 and 625, where the headerlet filename is <rootname>_flt_hlet.fits.

Reply:
Resolved by commit [5fc0e32](https://github.com/spacetelescope/hlapipeline/pull/56/commits/5fc0e324c29de0acb9706b50be7efcba5ce85711)